internal/services/azapi_resource_data_source.go

```diff
@@ -188,7 +192,18 @@ func (r *AzapiResourceDataSource) Read(ctx context.Context, request datasource.R
 		id = buildId
 	}
 
-	client := r.ProviderData.ResourceClient
+	var client clients.Requester
+	client = r.ProviderData.ResourceClient
+	if !model.RetryableErrors.IsNull() && !model.RetryableErrors.IsUnknown() {
+		bkof, regexps := clients.NewRetryableErrors(
+			model.RetryableErrors.GetIntervalSeconds(),
+			model.RetryableErrors.GetMaxIntervalSeconds(),
+			model.RetryableErrors.GetMultiplier(),
+			model.RetryableErrors.GetRandomizationFactor(),
```

Comment:
Same as above, because of the default value is not set, block like the following in a data source will cause panic:

```hcl
data "azapi_resource" "test" {
  type = "Microsoft.Resources/resourceGroups@2024-03-01"
  name = "henglu82"
  retryable_errors = {
    error_message_regex = [ "ResourceGroupNotFound" ]
  }
}
```